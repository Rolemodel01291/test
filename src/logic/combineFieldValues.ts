import { FieldValue } from '../types';

export default function combineFieldValues(data): FieldValue {
  const output = Object.entries(data).reduce((previous, [key, value]): FieldValue => {
    const arrayIndex = key.match(/\[\d+\]$/gi);

    if (arrayIndex) {
      const index = arrayIndex[0].slice(1, -1);
      const filedName = key.substr(0, key.indexOf('['));
      if (!previous[filedName]) previous[filedName] = [];
      previous[filedName][index] = value;
    } else {
      previous[key] = value;
    }
    return previous;
  }, {});

  return Object.entries(output).reduce((previous, [key, value]): FieldValue => {
    if (Array.isArray(value)) {
      previous[key] = value.filter(Boolean);
      return previous;
    }

    previous[key] = value;
    return previous;
  }, {});
}
