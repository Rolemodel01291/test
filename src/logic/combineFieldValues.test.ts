import combineFieldValues from './combineFieldValues';

describe('combineFieldValues', () => {
  it('should combine all the array fields', () => {
    expect(
      combineFieldValues({
        'email[1]': 'asdasd@dsad.com',
        'email[2]': 'asdasd@.com',
        'firstName[1]': 'asdasd',
        'firstName[2]': 'asdasd',
        'lastName[1]': 'asdasd',
        'lastName[2]': 'asd',
        test: 'test',
      }),
    ).toMatchSnapshot();
  });
});
