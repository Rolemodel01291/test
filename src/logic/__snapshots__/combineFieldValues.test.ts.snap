// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`combineFieldValues should combine all the array fields 1`] = `
Object {
  "email": Array [
    "asdasd@dsad.com",
    "asdasd@.com",
  ],
  "firstName": Array [
    "asdasd",
    "asdasd",
  ],
  "lastName": Array [
    "asdasd",
    "asd",
  ],
  "test": "test",
}
`;
